src/pysource/models.py

```diff
@@ -114,14 +120,20 @@ def __init__(self, origin, spacing, shape, space_order, nbl=20,
         self.grid = Grid(extent=extent, shape=shape_pml, origin=tuple(origin_pml),
                          dtype=dtype, subdomains=subdomains)
 
+        self._physical_parameters = []
+        # Q factor needed for damping initialization
+        if(Q is None):
+            self._Q_function = np.log(100)
+        else:
+            self._Q_function = self._gen_phys_param(Q, 'Q', space_order)
+
         if self.nbl != 0:
             # Create dampening field as symbol `damp`
             self.damp = Function(name="damp", grid=self.grid)
-            initialize_damp(self.damp, self.padsizes, spacing, fs=fs)
-            self._physical_parameters = ['damp']
+            initialize_damp(self.damp, self._Q_function, f0, self.padsizes, spacing, fs=fs)
```

Comment:
That's quite convoluted and force the creation of an extra function `Q` and of an unused `_physical_parameters`, `initialize_damp(self.damp, Q, ...)` should be enough